analyzer/src/main/kotlin/PackageCurationProvider.kt

```diff
@@ -33,14 +33,20 @@ interface PackageCurationProviderFactory<CONFIG> : ConfigurablePluginFactory<Pac
         val ALL = Plugin.getAll<PackageCurationProviderFactory<*>>()
 
         /**
-         * Return a new provider instance for each [enabled][PackageCurationProviderConfiguration.enabled] provider
-         * configuration in [configurations]. The given [configurations] must be ordered highest-priority first while
-         * the returned providers are ordered lowest-priority first, e.g. the order in which the corresponding curations
-         * must be applied.
+         * Return a new (identifier, provider instance) tuple for each
+         * [enabled][PackageCurationProviderConfiguration.enabled] provider configuration in [configurations]. The given
+         * [configurations] must be ordered highest-priority first while the returned providers are ordered
+         * lowest-priority first, e.g. the order in which the corresponding curations must be applied.
          */
-        fun create(configurations: List<PackageCurationProviderConfiguration>): List<PackageCurationProvider> =
-            configurations.filter { it.enabled }.map { ALL.getValue(it.type).create(it.config) }.asReversed()
-    }
+        fun create(
+            configurations: List<PackageCurationProviderConfiguration>
+        ): List<Pair<String, PackageCurationProvider>> =
+            configurations.filter {
+                it.enabled
+            }.map {
+                it.id to ALL.getValue(it.type).create(it.config)
+            }.asReversed()
```

Comment:
This function should throw an exception if there are two providers with the same `id`. It's better to have the check here than just in the `AnalyzerCommand` for reusability.